keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1196 @@
+
+```yaml
+kind: ResourceSlice
+...
+spec:
+  driver: cni.dra.networking.x-k8s.io
+  devices:
+  - name: eth1
+    basic:
+      shared: "true"
+      attributes:
+        name:
+          string: "eth1"
+      capacity:
+        bandwidth:
+          value: 10Gi
+- name: eth2
+    basic:
+      attributes:
+        name:
+          string: "eth1"
+      capacity:
+        bandwidth:
+          value: 10Gi
+---
+kind: DeviceClass
+metadata:
+  name: vlan-cni.networking.x-k8s.io
+```
+
+With the below ResourceClaim, either `eth1` or `eth2` can be selected. 
+The master field in the CNI config can be filled according to the selected device by the CNI DRA driver.
+
+```yaml
+kind: ResourceClaim
+...
+spec:
+  devices:
+    requests:
+    - name: macvlan
+      deviceClassName: vlan-cni.networking.x-k8s.io
+      resources:
+        requests:
+          bandwidth: "1Gi"
+    config:
+    - requests:
+      - macvlan
+      opaque:
+        driver: cni.dra.networking.x-k8s.io
+        parameters: # CNIParameters with the GVK, interface name and CNI Config (in YAML format).
+          apiVersion: cni.networking.x-k8s.io/v1alpha1
+          kind: CNI
+          ifName: "net1"
+          config:
+            cniVersion: 1.0.0
+            name: net1
+            plugins:
+            - type: macvlan
+              mode: bridge
+              ipam:
+                type: host-local
+                ranges:
+                - - subnet: 10.10.1.0/24
+```
+
+#### Story 2
+A user reserves that fractions of memory for the virtual GPU.
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaim
+...
+spec:
+  devices:
+    requests:
+    - name: vgpu0
+      deviceClassName: vgpu.nvidia.com
+      resources:
+        requests:
+          memory: "10Gi"
+```
+
+#### Story 3
+A user specifies a sharted device by some attributes such as name.
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaim
+...
+spec:
+  devices:
+    requests:
+    - name: net1
+      deviceClassName: vlan-cni.networking.x-k8s.io
+      selectors:
+      - cel:
+          expression: "device.attributes['vlan-cni.networking.x-k8s.io'].name == 'eth1'"
+      resources:
+        requests:
+          bandwidth: "1Gi"
+```
+
+#### Story 4
+A user reserves the shared device and blocks the others to get the same shared device.
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaim
+...
+spec:
+  devices:
+    requests:
+    - name: net1
+      deviceClassName: vlan-cni.networking.x-k8s.io
+      resources:
+        all: "true"
+```
+
+#### Story 5
+A user defines resource limits when the device driver supports a burstable use.
+The CNI DRA driver can add a chained bandwidth CNI based on the bandwidth request.
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaim
+...
+spec:
+  devices:
+    requests:
+    - name: net1
+      deviceClassName: vlan-cni.networking.x-k8s.io
+      resources:
+        requests:
```

Comment:
We should probably also clarify here how shared devices works with the abstract pools introduced in https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/4815-dra-partitionable-devices. If part of a shared device is allocated, I would assume that means that all the `consumesCapacity` specified for the device would be considered "used" by the capacity pool?

Reply:
Yes. if any part of a partition is used, then that partition is in-use and allocated

Reply:
I have updated the [KEP design detail's scheduling enhancement](https://github.com/sunya-ch/k8s-enhancements/tree/dra-dynamic-provision/keps/sig-scheduling/5075-dra-consumable-capacity#scheduling-enhancement). The shared device will not included in allocatedDevice and allocatingDevice to allow multiple-claim allocation. Additionally, with consumable capacity, scheduler will handle allocated devices with allocatedCapacity and allocatingCapacity instead.